packages/rx_bloc/test/bloc/counter_test.dart

```diff
@@ -35,125 +36,145 @@ void main() {
   });
 
   group('CounterBloc: isLoading', () {
-    rxBlocTest<CounterBlocType, bool>(
+    rxBlocTest<CounterBlocType, LoadingWithTag>(
       'CounterBloc: dispose',
-      build: () async => CounterBloc(ServerSimulator()),
+      build: () async => CounterBloc(CounterRepository()),
       state: (bloc) => bloc.states.isLoading,
       act: (bloc) async => bloc
         ..events.increment()
         ..dispose(),
-      wait: ServerSimulator.delayTest,
-      expect: <bool>[false],
+      expect: <LoadingWithTag>[LoadingWithTag(loading: false)],
     );
 
-    rxBlocTest<CounterBlocType, bool>(
+    rxBlocTest<CounterBlocType, LoadingWithTag>(
       'CounterBloc: isLoading: no count subscription',
-      build: () async => CounterBloc(ServerSimulator()),
+      build: () async => CounterBloc(CounterRepository()),
       state: (bloc) => bloc.states.isLoading,
       act: (bloc) async => bloc.events.increment(),
-      wait: ServerSimulator.delayTest,
-      expect: <bool>[false],
+      expect: <LoadingWithTag>[LoadingWithTag(loading: false)],
     );
 
-    rxBlocTest<CounterBlocType, bool>(
+    rxBlocTest<CounterBlocType, LoadingWithTag>(
       'CounterBloc: isLoading one increment event',
-      build: () async => CounterBloc(ServerSimulator()),
+      build: () async => CounterBloc(CounterRepository()),
       state: (bloc) => bloc.states.isLoading,
       act: (bloc) async {
         bloc.states.count.listen((event) {});
         bloc.events.increment();
       },
-      wait: ServerSimulator.delayTest,
-      expect: <bool>[false, true, false],
+      expect: <LoadingWithTag>[
+        LoadingWithTag(loading: false),
+        LoadingWithTag(loading: true, tag: CounterBloc.incrementTag),
+        LoadingWithTag(loading: false, tag: CounterBloc.incrementTag),
+      ],
     );
 
-    rxBlocTest<CounterBlocType, bool>(
-      'CounterBloc: isLoading one decrement event',
-      build: () async => CounterBloc(ServerSimulator()),
+    rxBlocTest<CounterBlocType, LoadingWithTag>(
+      'CounterBloc - isLoading one decrement event',
+      build: () async => CounterBloc(CounterRepository()),
       state: (bloc) => bloc.states.isLoading,
       act: (bloc) async {
         bloc.states.count.listen((event) {});
         bloc.events.decrement();
       },
-      wait: ServerSimulator.delayTest,
-      expect: <bool>[false, true, false],
+      expect: <LoadingWithTag>[
+        LoadingWithTag(loading: false),
+        LoadingWithTag(loading: true, tag: CounterBloc.decrementTag),
+        LoadingWithTag(loading: false, tag: CounterBloc.decrementTag),
+      ],
     );
 
-    rxBlocTest<CounterBlocType, bool>(
-      'CounterBloc: isLoading two increment event',
-      build: () async => CounterBloc(ServerSimulator()),
+    rxBlocTest<CounterBlocType, LoadingWithTag>(
```

Comment:
@PrimeSwiftArt, do you think of any other possible scenario that we need to test?